libs/knowledge-store/ragstack_knowledge_store/embedding_model.py

```diff
@@ -1,22 +1,67 @@
-from abc import ABC, abstractmethod
-from typing import List
-
+from abc import ABC
+from typing import List, Any, Optional
 
 class EmbeddingModel(ABC):
     """Embedding model."""
 
-    @abstractmethod
+    def __init__(self, embeddings: Any, method_map: Optional[dict] = None, other_methods: Optional[List[str]] = None):
+        self.embeddings = embeddings
+        self.method_name = {}
+        method_map = method_map if method_map else {}
+        other_methods = other_methods if other_methods else []
+
+        base_methods = ['embed_texts', 'aembed_texts', 'embed_query', 'aembed_query']
```

Comment:
I don't think we should try to add all of these as methods, it's definitely pretty messy.

I think we should just have `embed_mime(self, mime_type: str, content: Union[str, Bytes])` or something like that. Then there is only a single abstract method to use for any mime type and the names can be different, etc.

Reply:
100% but right now, LangChain doesn't have "embed_mime" :)